src/ert/parsing/new_observations_parser.py

```diff
@@ -48,21 +37,420 @@ def from_rule(cls, rule: str) -> "ObservationType":
+            SimpleHistoryDeclaration,
+            Tuple[ObservationType, FileContextToken, Dict[FileContextToken, Any]],
+        ]
+    ],
+):
+    result = []
+    for decl in inp:
+        if decl[0] == ObservationType.HISTORY:
+            if len(decl) == 2:
+                result.append((decl[0], decl[1], _validate_history_values(decl[1], {})))
+            if len(decl) == 3:
+                result.append(
+                    (decl[0], decl[1], _validate_history_values(decl[1], decl[2]))
+                )
+        elif decl[0] == ObservationType.SUMMARY:
+            if len(decl) != 3:
+                raise _unknown_declaration_error(decl)
+            result.append(
+                (decl[0], decl[1], _validate_summary_values(decl[1], decl[2]))
+            )
+        elif decl[0] == ObservationType.GENERAL:
+            if len(decl) != 3:
+                raise _unknown_declaration_error(decl)
+            result.append(
+                (
+                    decl[0],
+                    decl[1],
+                    _validate_gen_obs_values(directory, decl[1], decl[2]),
+                )
+            )
+        else:
+            raise _unknown_declaration_error(decl)
+    return result
+
+
+def _validate_history_values(
+    name_token: FileContextToken, inp: Dict[FileContextToken, Any]
+) -> HistoryValues:
+    error_mode = "RELMIN"
+    error = 0.1
+    error_min = 0.1
+    segment = []
+    for key, value in inp.items():
+        if key == "ERROR":
+            error = validate_positive_float(value, key)
+        elif key == "ERROR_MIN":
+            error_min = validate_positive_float(value, key)
+        elif key == "ERROR_MODE":
+            error_mode = validate_error_mode(value)
+        elif key == "SEGMENT":
+            segment.append((value[0], _validate_segment_dict(key, value[1])))
+        else:
+            raise _unknown_key_error(key, name_token)
+
+    return {
+        "ERROR_MODE": error_mode,
+        "ERROR": error,
+        "ERROR_MIN": error_min,
+        "SEGMENT": segment,
+    }
+
+
+def _validate_summary_values(
+    name_token: FileContextToken, inp: Dict[FileContextToken, Any]
+) -> SummaryValues:
+    error_mode = "ABS"
+    summary_key = None
+
+    date_dict: DateDict = {}
+    float_values: Dict[str, float] = {"ERROR_MIN": 0.1}
+    for key, value in inp.items():
+        if key == "RESTART":
+            date_dict["RESTART"] = validate_positive_int(value, key)
+        elif key in ["ERROR", "ERROR_MIN"]:
+            float_values[str(key)] = validate_positive_float(value, key)
+        elif key in ["DAYS", "HOURS"]:
+            date_dict[str(key)] = validate_positive_float(value, key)
+        elif key == "VALUE":
+            float_values["VALUE"] = validate_float(value, key)
+        elif key == "ERROR_MODE":
+            error_mode = validate_error_mode(value)
+        elif key == "KEY":
+            summary_key = value
+        elif key == "DATE":
+            date_dict["DATE"] = value
```

Comment:
Replace lookup strings with key? There are some other lookups that could be changed elsewhere as well.

Reply:
Should be fixed now.